src/rabbit_fifo.erl

```diff
@@ -232,7 +232,12 @@
          prefix_msg_counts = {0, 0} :: {Return :: non_neg_integer(),
                                         PrefixMsgs :: non_neg_integer()},
          msg_bytes_enqueue = 0 :: non_neg_integer(),
-         msg_bytes_checkout = 0 :: non_neg_integer()
+         msg_bytes_checkout = 0 :: non_neg_integer(),
+         %% whether single active consumer is on or not for this queue
+         single_active_consumer_on = false :: boolean(),
+         %% waiting consumers, one is picked active consumer is cancelled or dies
+         %% used only when single active consumer is on
+         waiting_consumers = [] :: list()
```

Comment:
should this be `waiting_consumers = [] :: [{consumer_id(), consumer()}]` ?

Reply:
Done. This needed to declare a new `consumer` type, just after the record declaration.